### PR TITLE
Add flag to enable running tests with Fiddler

### LINKED ISSUE #27316: [BUG] New .NET test framework does not run tests in Record or Playback mode with Fiddler running.
### Library name and version

Azure.Core.TestFramework

### Describe the bug

New .NET test framework does run tests in Record or Playback mode with Fiddler running.

Work around:

https://github.com/Azure/azure-sdk-for-net/blob/edee9bbff84e4cf09091e3e0537d72b10e2bee88/sdk/core/Azure.Core.TestFramework/src/ProxyTransport.cs#L163
Updated to `request.Uri.Host = "localhost.fiddler"; `
https://github.com/Azure/azure-sdk-for-net/blob/edee9bbff84e4cf09091e3e0537d72b10e2bee88/sdk/core/Azure.Core.TestFramework/src/TestProxy.cs#L27
Update to `public const string DevCertIssuer =  "CN=DO_NOT_TRUST_FiddlerRoot, O=DO_NOT_TRUST, OU=Created by http://www.fiddler2.com/";` 

### Expected behavior

I expect to be able to run tests and Fiddler at the same time.

### Actual behavior

I am not able to run tests in Playback or Record mode with Fiddler running.

### Reproduction Steps

Run tests in Playback or Record mode with Fiddler enabled.

### Environment

Windows 10.

### PR DESCRIPTION
**Pull request by JoshLove-msft**
Fixes https://github.com/Azure/azure-sdk-for-net/issues/27316

### CODE REVIEW

File: sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs [hunk 1]
```diff
@@ -154,6 +154,12 @@ public string ReplacementHost
         {
         };
 
+        /// <summary>
+        /// Whether or not to enable the test framework to proxy traffic through fiddler. This should only be set to
+        /// True for local debugging scenarios, when you have Fiddler listening for traffic.
+        /// </summary>
+        public bool EnableFiddler { get; set; }
```
Comment (christothes) [inline sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs:161]: Can we read it from the environment also? That would make it less likely to check in.

Comment (JoshLove-msft) [reply sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs:161]: I'd rather not add the overhead/edge cases that comes along with trying to detect this automatically. There is no danger if someone checks it in - it is similar to accidentally checking in RecordedTestMode.Record - the CI will fail.

Comment (christothes) [reply sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs:161]: I think we should avoid popular config options that are easy to check in and ruin a CI run.

Comment (JoshLove-msft) [reply sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs:161]: I don't think this will be a popular option.

Comment (christothes) [reply sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs:161]: It will be for me 😄

Comment (JoshLove-msft) [reply sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs:161]: Oh you mean read it from like a flag that someone sets in the env?

Comment (JoshLove-msft) [reply sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs:161]: Ok I think that is reasonable. @seanmcc-msft any concerns if we implement this as an env var that you would set?

Comment (seanmcc-msft) [reply sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs]: No that is fine.

**Review by seanmcc-msft — APPROVED** [review]

**Pull request merged**